### PR TITLE
10159 android m port

### PR DESCRIPTION
**Pull request by shanetmoore**
Port Android M changes made on 4.0.3 up to Android 4.1

### CODE REVIEW

**Comment by shanetmoore** [comment]
@Sergio-Mira @dominicmarmionswrve

File: SwrveSDKDemo/build.gradle [hunk 1]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 22
-    buildToolsVersion '22.0.1'
+    compileSdkVersion 23
+    buildToolsVersion '21.1.2'
```
Comment (Sergio-Mira) [inline SwrveSDKDemo/build.gradle]: Why downgrading the build tools?

**Comment by Sergio-Mira** [comment]
:rocket:

File: SwrveSDKDemo/build.gradle (hunk 1 above)
Comment (shanetmoore) [inline SwrveSDKDemo/build.gradle]: Yeah Dom had said he was having issues. I'd like to use 23.01 across the board. 

I'll make a point to upgrade them today and jira up any issues I come across

**Pull request merged**